Runar/Runar/Services/Notifications/LocalNotificationManager.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  LocalNotificationManager.swift
+//  Runar
+//
+//  Created by Alexey Poletaev on 09.02.2023.
+//
+
+import UserNotifications
+
+private extension String {
+    static let notificationGeneralContentTitle = L10n.Notifications.General.title
+    static let notificationGeneralContentBody = L10n.Notifications.General.body
+    static let notificationGeneralIdentifier = "General Local Notification"
+}
+
+private enum TimeOfSendingLocalNotifications: Int {
+    case weekday = 2 // Monday
+    case hour = 11   // 11:00
+}
+
+protocol LocalNotificationInterface {
+    /// Checking the resolution of notifications and add notification
+    func addNotification()
+
+    /// Remove all notification
+    func removeAllNotifications()
+}
+
+final class LocalNotificationManager: NSObject, LocalNotificationInterface {
+
+    private let notificationCenter = UNUserNotificationCenter.current()
+
+    /// If the user logs into the app on the day of the notification call before the time of the call, the notification on that day will be deleted
+    private func checkTheDayAndTime() {
+        let dateComponents = Calendar.current.dateComponents([.weekday, .hour, .weekOfMonth], from: Date())
+        let weekdaySetPoint = TimeOfSendingLocalNotifications.weekday.rawValue
+        let hourSetPoint = TimeOfSendingLocalNotifications.hour.rawValue
+
+        guard let weekday = dateComponents.weekday,
+              let hour = dateComponents.hour,
+              let week = dateComponents.weekOfMonth else { return }
+
+        if weekday == weekdaySetPoint && hour < hourSetPoint {
```

Comment:
Could you explain in simple this domain logic?

Reply:
When user open application, we get the current weekday and hour and compare them with the setpoint for notification.